src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -318,3 +312,42 @@ async def get_metric_value(
             )
 
         return metric_values
+
+
+class CompileWaitingTimeMetricsCollector(MetricsCollector):
+    """
+    This Metric will track the amount of time compile requests spends waiting in the compile queue before being executed.
+    """
+
+    def get_metric_name(self) -> str:
+        return "orchestrator.compile_waiting_time"
+
+    def get_metric_type(self) -> MetricType:
+        return MetricType.TIMER
+
+    async def get_metric_value(
+        self, start_interval: datetime, end_interval: datetime, connection: asyncpg.connection.Connection
+    ) -> Sequence[MetricValueTimer]:
+        query: str = f"""
+            SELECT count(*)  as count,environment,sum(started-requested) as compile_waiting_time
+            FROM {Compile.table_name()}
+            WHERE started >= '{start_interval}'
+            AND started < '{end_interval}'
+            GROUP BY environment
```

Comment:
We should never use string concatenation in queries. This opens the door for SQL-injection attacks.

Reply:
Nice catch, @FloLey can you also fix this in the compile time metrics query ?
How can we fix this, should we use query parameters instead ?

Reply:
Yes. You write the query with `$<number>` placeholders in the locations where you have a parameter. Asyncpg will then make sure that the parameter is correctly escaped to prevent an incorrect interpretation of the value. This is not possible for the `{Compile.table_name()}` substitution, but that is not bad because the value is directly derived from the name of a class. So we know it's harmless.